x-pack/plugins/cases/server/routes/api/register_routes.ts

```diff
@@ -71,15 +71,18 @@ const logAndIncreaseDeprecationTelemetryCounters = ({
 };
 
 export const registerRoutes = (deps: RegisterRoutesDeps) => {
-  const { router, routes, logger, kibanaVersion, telemetryUsageCounter } = deps;
+  const { router, routes, logger, kibanaVersion, telemetryUsageCounter, isServerless } = deps;
 
   routes.forEach((route) => {
     const { method, path, params, options, routerOptions, handler } = route;
 
+    const routeAccess =
+      isServerless && options?.deprecated ? 'internal' : routerOptions?.access ?? 'internal';
+
     (router[method] as RouteRegistrar<typeof method, CasesRequestHandlerContext>)(
       {
         path,
-        options: routerOptions,
+        options: { ...routerOptions, access: routeAccess },
```

Comment:
I don't have a better solution but I don't like that we have `get_external_routes` and `get_internal_routes` but then there is this hack when registering the routes that makes that logic irrelevant.

They are deprecated anyway so it should be fine but kinda feels like code smell 😅